pkg/cortex/cortex.go

```diff
@@ -293,11 +296,15 @@ func New(cfg Config) (*Cortex, error) {
 
 	// Don't check auth header on TransferChunks, as we weren't originally
 	// sending it and this could cause transfers to fail on update.
-	//
-	// Also don't check auth /frontend.Frontend/Process, as this handles
-	// queries for multiple users.
 	cfg.API.HTTPAuthMiddleware = fakeauth.SetupAuthMiddleware(&cfg.Server, cfg.AuthEnabled,
-		[]string{"/cortex.Ingester/TransferChunks", "/frontend.Frontend/Process"})
+		// Also don't check auth for these gRPC methods, since single call is used for multiple users (or no user like health check).
+		[]string{
+			"/grpc.health.v1.Health/Check",
```

Comment:
Client pool uses health check to remote stale clients from the pool. When issuing Check method, it sets user "0", to avoid authentication errors. (https://github.com/cortexproject/cortex/blob/4b40f20aa8f0d0bead9e65e6a25fdee78a8787a3/pkg/ring/client/pool.go#L193)

I've tried to remove that setting of "0" and add this exception instead. It worked fine between querier and frontend, but not between querier and ingester, simply because I haven't deployed updated ingester during my testing. I have then realized that it would be bit painful to migrate if I had removed that setting of "0" user, so first step in migration is just adding this exception.

That said, we can remove it from this PR as it's currently not needed.